gcc/rust/parse/rust-parse-impl.h

```diff
@@ -793,6 +793,20 @@ Parser<ManagedTokenSource>::parse_attr_input ()
 
 	t = lexer.peek_token ();
 
+	// attempt to parse macro
+	// TODO: macros may/may not be allowed in attributes
+	// this is needed for "#[doc = !include_str(...)]"
+	if (is_simple_path_segment (t->get_id ()))
```

Comment:
we can have more complex macro invocations than single paths here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d81ca952b1610239de6a3955dd175728

```rust
#[macro_export]
macro_rules! foo {
    () => { "hello" }
}

#[doc = crate::foo!()]
struct S;
```

Reply:
I can change it, but ```parse_macro_invocation``` currently only accepts simple paths

https://github.com/Rust-GCC/gccrs/blob/master/gcc/rust/parse/rust-parse-impl.h#L1845

Reply:
Oh, wait, *single* paths

Reply:
Oh, actually this might be my mistake. The snippet I put in my comment is already parsed properly by your fix, so that's all good. I'm the one that got confused by "simple/single"!